src/pages/api/v3/measurements/index.ts

```diff
@@ -220,128 +220,128 @@ const handler = async (req: NextApiRequest, res: NextApiResponse) => {
-      } else {
-        /* if no location is found nearby, create a new location whose name could be filled in later, useful when on boat? */
-        const DEFAULT_RADIUS = 200;
-        locationId = await Location.createOne({
-          name: "unknown",
-          long: position.long,
-          lat: position.lat,
-          rad: DEFAULT_RADIUS,
-        });
-      }
+    for (const body of arrayedBody) {
+      try {
+        /* validate request body */
+        const { time, position, sensors } = zCreateMeasurement.parse(body);
 
-      let insertedMeasurements = [];
-      let errors: { sensorId: number; status: string }[] = [];
-      for (const { id, value, unit } of sensors) {
-        try {
-          /* find associated sensor by the id to get its type */
-          const sensor = await Sensor.findById({ id });
-          if (!sensor) {
-            errors.push({ sensorId: id, status: "SENSOR_NOT_FOUND" });
-          }
+        /* find location in the db from the lat,long specified */
+        const closestLocation = await Location.findClosest(position);
+        let locationId: number;
+        if (closestLocation) {
+          locationId = closestLocation.id;
+        } else {
+          /* if no location is found nearby, create a new location whose name could be filled in later, useful when on boat? */
+          const DEFAULT_RADIUS = 200;
+          locationId = await Location.createOne({
+            name: `unknown-${Math.floor(Math.random() * 100)}`,
+            long: position.long,
+            lat: position.lat,
+            rad: DEFAULT_RADIUS,
+          });
+        }
 
-          /* convert the value to SI-unit if there is one */
-          let convertedValue = value;
-          if (sensor.type === "temperature") {
-            convertedValue = parseTemperature(value, unit || "k").asKelvin();
-          } else if (sensor.type === "conductivity") {
-            convertedValue = parseConductivity(
-              value,
-              unit || "spm"
-            ).asSiemensPerMeter();
-          } else if (sensor.type === "ph") {
-            convertedValue = new PH(value).getValue();
-          } else {
-            convertedValue = round(value);
-          }
+        for (const { id, value, unit } of sensors) {
+          try {
+            /* find associated sensor by the id to get its type */
+            const sensor = await Sensor.findById({ id });
+            if (!sensor) {
+              sensorErrors.push({ sensorId: id, status: "SENSOR_NOT_FOUND" });
+              continue;
+            }
 
-          /* insert measurement into db */
-          await Measurement.createOne({
-            sensorId: id,
-            value: convertedValue,
-            time,
-            sensorType: sensor.type,
-            locationId,
-            position,
-          });
+            /* convert the value to SI-unit if there is one */
+            let convertedValue = value;
+            if (sensor.type === "temperature") {
```

Comment:
Why is there a comparison with a hand typed string if classes/types for temperature and other units exist?
I expected something like this:
```js
switch(sensore.type) {
case Temperature.name: doTempStuff
case Ph.name: doPhStuff
}
```